packages/@glimmer/runtime/lib/vm/rehydrate-builder.ts

```diff
@@ -27,8 +27,16 @@ export class RehydrateBuilder extends NewElementBuilder implements ElementBuilde
   constructor(env: Environment, parentNode: Simple.Element, nextSibling: Option<Simple.Node>) {
     super(env, parentNode, nextSibling);
     if (nextSibling) throw new Error("Rehydration with nextSibling not supported");
-    this.candidate = this.currentCursor!.element.firstChild;
-    assert(this.candidate && isComment(this.candidate) && this.candidate.nodeValue === '%+b:0%', 'Must have opening comment <!--%+b:0%--> for rehydration.');
+
+    let node = this.currentCursor!.element.firstChild;
+
+    while (node) {
+      if (node && isComment(node) && node.nodeValue === '%+b:0%') { break; }
+      node = node.nextSibling;
+    }
+
+    this.candidate = node;
```

Comment:
I believe setting `this.candidate` to the found `node` is correct.  Since we know the rehydration builder should not need to touch the nodes before the openning comment marker is set.